exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetReaderUtility.java

```diff
@@ -214,14 +218,24 @@ public static DateCorruptionStatus detectCorruptDates(ParquetMetadata footer,
           }
           // written by a tool that wasn't Drill, the dates are not corrupted
           return DateCorruptionStatus.META_SHOWS_NO_CORRUPTION;
-        } catch (VersionParser.VersionParseException e) {
-          // If we couldn't parse "created by" field, check column metadata of date columns
-          return checkForCorruptDateValuesInStatistics(footer, columns, autoCorrectCorruptDates);
         }
       }
+    } catch (VersionParser.VersionParseException e) {
+      // If we couldn't parse "created by" or "drill version", check column metadata of date columns
+      return checkForCorruptDateValuesInStatistics(footer, columns, autoCorrectCorruptDates);
     }
   }
 
+  public static boolean isDrillVersionHasCorrectDates(String drillVersion) throws VersionParser.VersionParseException {
```

Comment:
Might flow better as "drillVersionHasCorrectDates"

But, see comment above about whether we need this check.